### PR TITLE
PropertiesModule: Print properties, processors, totalMemory on startup.

### PR DESCRIPTION
**Pull request by gianm**
Should help with debugging issues from the mailing lists.

### CODE REVIEW

**Comment by fjy** [comment]
:+1:

**Comment by fjy** [comment]
@gianm can we backport this?

File: processing/src/main/java/io/druid/guice/PropertiesModule.java [hunk 1]
```diff
@@ -86,6 +87,16 @@ public void configure(Binder binder)
...
     }
 
+    log.info(
+        "Loaded properties into JVM with processors[%,d], memory[%,d].",
+        Runtime.getRuntime().availableProcessors(),
+        Runtime.getRuntime().totalMemory()
+    );
+
+    for (String propertyName : Ordering.natural().sortedCopy(props.stringPropertyNames())) {
+      log.info("* %s: %s", propertyName, props.getProperty(propertyName));
```
Comment (drcrallen) [inline processing/src/main/java/io/druid/guice/PropertiesModule.java]: this will print aws credentials in log, please remove.

Comment (drcrallen) [inline processing/src/main/java/io/druid/guice/PropertiesModule.java]: same with db user pass

Comment (drcrallen) [inline processing/src/main/java/io/druid/guice/PropertiesModule.java]: IMHO printing out the property names should be fine, but we should not print out property values unless we have a way to differentiate secure from insecure properties.

Comment (fjy) [inline processing/src/main/java/io/druid/guice/PropertiesModule.java]: you don't have to put your properties there, there are ways to read these credentials from file

Comment (drcrallen) [inline processing/src/main/java/io/druid/guice/PropertiesModule.java]: Then we should remove support for the way that puts plain text credentials into the log by default.

**Comment by drcrallen** [comment]
@gianm acceptable alternatives would make this configurable as opt-in, or simply printing out the keys but not the values.

File: processing/src/main/java/io/druid/guice/PropertiesModule.java (hunk 1 above)
Comment (nishantmonu51) [inline processing/src/main/java/io/druid/guice/PropertiesModule.java]: druid does not provide any default way to supply the db credentials other than passing them in properties file.

**Comment by nishantmonu51** [comment]
+1, on making it configurable.

**Comment by drcrallen** [comment]
@gianm I'd be ok with making the setting to enable it have it set equal to TRUE in the template runtime.properties that get packaged by default, but in the absence of a setting printing of the value of properties should be disabled.

**Comment by gianm** [comment]
@drcrallen sounds good, will do another PR.

**Pull request merged**